Update action everytime

srisankethu opened the pull request:
Fixes https://github.com/coala-analyzer/coala/issues/1825
Takes in `failed_actions` and asks  for related params in case of a failed execution

tushar-rishav commented:
@srisankethu Seems like you need to:

- Add tests for the changes. That'd probably clear the tests and coverage.
 
- Fix commit. Please follow the commit guidelines.

-  Finally, please run tests locally before sending PR.

sudheesh001 commented:
unack d716662

srisankethu commented:
@tushar-rishav The PR solves the issue( as far as I have checked). Just need to add tests

sils1297 commented:
hm @srisankethu it's definitely going in the right direction now, I wonder if we could do something simpler to avoid passing down the whole failed actions list everywhere...

srisankethu commented:
@sils1297 failed_actions moves around 3 functions only ! I think it is simple enough as the problem was in appending section, so we need it in get_action_info and we need to check it in ask_action _and_apply.

abhsag24 commented:
unack  89cd6b4

abhsag24 commented:
some minor changes and a rebase, though i like where this is going :)

sils1297 commented:
unack e563cd3, but we're getting to it indeed

sils1297 commented:
@srisankethu seems test coverage is insufficient

tushar-rishav commented:
@srisankethu  This is annoying. Would you *please* do these two things before each PR.

- Run `py.test`
- Run `coala`
*Please*?

tushar-rishav commented:
unack a17a538

srisankethu commented:
@tushar-rishav I'm sorry if you're feeling annoyed. My tests did a 100% coverage earlier, but @sils1297 pointed out a test case with he thought was unnecessary. So I just removed it, and made a slight change to the other test cases to cover what that test case was covering. It missed an if-condition(...again). So now I have added my previous test case and corrected the test cases according to all the reviews given.

AbdealiJK commented:
@srisankethu seems like coverage missed again - so marking as wip

srisankethu commented:
@AbdealiJK Do you any other changes in my PR, it kinda frustating make changes continuously. I hope you understand.

AbdealiJK commented:
@srisankethu sure - let me do a quick review then

srisankethu commented:
@AbdealiJK Yes I was thinking about not adding comments to the test because I didn't find any test file having comments.

sils1297 commented:
@srisankethu hey, I can understand that code review can sometimes be frustrating - we know it well, I worked once for months on one PR with @Makman2...

srisankethu commented:
@sils1297 Wow! Yes, deep down i understand that it's a way to learn and correct yourself :+1:

srisankethu commented:
Updated changes :)

sils1297 commented:
see old review on https://github.com/coala-analyzer/coala/commit/51c4fe686874931a0ed60edd6e55e699418bae7e

sils1297 commented:
unack https://github.com/srisankethu/coala/commit/b3dc441d621f5f84a1f36805f56e48bfb6e8c582 as per previous review

sils1297 commented:
please read *all* comments from the previous review, correcting some isn't enough and is just more work for both of us.

srisankethu commented:
@sils1297 Im sorry , I was in a hurry, gotta go for lunch

arafsheikh commented on coalib/output/ConsoleInteraction.py:
```diff
@@ -453,13 +455,16 @@ def acquire_settings(log_printer, settings_names_dict):
 
     :param section: The section the action corresponds to.
     :param action:  The action to get the info for.
+    :param failed_actions:  A set of all actions that have failed.A failed
```
Alignment problem. `The action...` should be exactly below `A set...`

srisankethu replied:
OK !

sils1297 commented:
tests fail

sils1297 commented:
unack https://github.com/srisankethu/coala/commit/db299ef8c986ae68d5b3b3999e976da63b293f47

srisankethu commented:
Done :+1:

sils1297 commented:
unack e519580f271f5b90495658ebcb1ac7c0a34569b7

sils1297 commented:
@rultor merge

rultor commented:
> @rultor merge

@sils1297 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/7647-195743350)

rultor commented:
> @rultor merge

@sils1297 Done! FYI, the full log is [here](http://www.rultor.com/t/7647-195743350) (took me 1min)
